src/global_distrib/mod_global_distrib_outgoing_conns_sup.erl

```diff
@@ -33,10 +33,7 @@ add_server(Server) ->
 
 -spec get_connection(Server :: ejabberd:lserver()) -> pid().
 get_connection(Server) ->
-    case whereis(mod_global_distrib_utils:server_to_sup_name(Server)) of
-        undefined -> ok = add_server(Server);
-        _ -> ok
-    end,
+    ensure_server_started(Server),
```

Comment:
What if there are several processes calling this get_connection at the same time?

Reply:
Why should that be a problem?

Reply:
Maybe because of this?
https://github.com/esl/MongooseIM/pull/1660/files#r165373503
I know, it's very improbable to get this bug in tests, but in the real world (c) we often can get 10k routing calls at the same time. or 100k+ over several minutes.

Reply:
ah, so easy to reproduce the bug.
Two routes from two processes.
One creates a supervisor of connections. gen_tcp:connect can block init/1 function for some time.
So, second process would fail inside ensure_started.
at least it's not possible to start several supervisors, because the main supervisor would not allow it (but again, exactly that's why there is a badmatch in easure started).